enhancements/machine-config/install-time-support-for-image-mode-on-openshift.md

```diff
@@ -0,0 +1,342 @@
+
+OpenShift's current approach to OS management provides excellent consistency and supportability through RHCOS (Red Hat CoreOS), but it requires cluster admins to cede certain configurability aspects to the platform. As workloads become more specialized there is an increasing need for OS-level customization that can be applied from Day 0. An obvious example today is how AI workloads require specific hardware drivers and configurations on nodes.
+
+Some current challenges that Install Time Support For Image Mode on OpenShift would solve:
+- Inability to customize OS images during initial cluster installation
+- Post-installation customization complexity and timing issues
+- Difficulty ensuring all nodes (including control plane nodes) have consistent custom configurations
+- Limited support for specialized hardware requirements during installation
+
+### User Stories
+
+#### Story 1: AI/ML Workload Deployment
+
+As a cluster admin deploying AI workloads, I want to include specialized GPU drivers and AI frameworks in my OS image at install time, so that my cluster is ready for AI workloads from the moment it comes online, without requiring post-installation configuration.
+
+#### Story 2: Regulated Environment Deployment
+
+As a cluster admin in a regulated environment, I want to include security agents and compliance tools in my OS image during installation, so that my cluster meets compliance requirements from the moment it comes online, without requiring post-installation configuration.
+
+#### Story 3: Edge Computing Deployment
+
+As a cluster admin deploying edge computing solutions, I want to include edge-specific drivers and monitoring agents in my OS image during installation, so that my edge nodes are immediately functional without requiring network connectivity for post-installation customization.
+
+### Goals
+
+- Enable cluster admins to use Image Mode on OpenShift at install time
+- Support custom OS image builds for master, worker, and any other custom pools during bootstrap
+- Maintain OpenShift's single-click upgrade experience maintaining customizations across upgrades
+- Ensure control plane and worker nodes can all use custom OS images
+- Provide a declarative approach to OS customization through MachineOSConfig CRD
+
+### Non-Goals
+
+- Replacing the existing ignition-based configuration system entirely
+- Supporting arbitrary OS distributions beyond RHCOS-based images
+- Providing external CI/CD integration for image builds (builds occur on-cluster)
+
+## Proposal
+
+### Workflow Description
+
+The install time Image Mode on OpenShift workflow integrates with the existing OpenShift installation process in the following steps.
+
+#### Step 1: Installation Configuration
+
+1. Cluster admin runs the installer (openshift-install) to generate install time manifests
+2. Cluster admin specifies "on-cluster" image mode enablement in `install-config.yaml`
```

Comment:
I could be mis-remembering, but I was under the impression that we didn't need an explicit flag for this, but rather the existence of MOSC objects in the install manifests would point to enabled OCL.

Also, are we planning to allow the user to e.g. have non-image mode control plane and image-mode workers during install?